amqp/exchangePublisher.go

```diff
@@ -52,3 +52,21 @@ func (p *ExchangePublisher) Publish(exchangeName string, headers map[string]stri
 
 	return nil
 }
+
+// PublishWithRoutingKey publishes a message to the given exchange, with a routing key to specify the queue
+func (p *ExchangePublisher) PublishWithRoutingKey(exchangeName string, routingKey string, body []byte) error {
```

Comment:
If it's reasonable within your timelines, can you make this function support headers and them use it in the `Publish()` signature above? e.g.

```go
func (p *ExchangePublisher) PublishWithRoutingKey(exchangeName string, routingKey string, headers map[string]string, body []byte) error {
    /* snip */
}

func (p *ExchangePublisher) Publish(exchangeName string, headers map[string]string, body []byte) error {
    return PublishWithRoutingKey(exchangeName, "", headers, body)
}
```

Reply:
Sure I can do that. My original intention was to have `Publish` and `PublishWithRoutingKey` mimic, respectively, the signatures of the Queue and PubSub implementations from the Axis library. In the Axis Queue implementation, the headers are used to overload the (single) queue to specify different message types (with no routing key), whereas Pubsub uses a single message type (i.e. no headers) but routing keys to go to many queues.
So ultimately, anything internal to Syncromatics that would use this lib for Pubsub would pass in an empty headers map. I'm slightly inclined to make a breaking change and fold them together into a single signature. Clients would either pass in an empty routing key or an empty headers map.

Reply:
I'm okay with either approach: two signatures with the uses you described OR making a breaking change. If you do the breaking change, just make sure to update the module version too.

Reply:
Ok thanks. My top preference is to leave it as written above, since that matches the existing use cases in our system. I know the function names are a bit clunky with no overloading in go, but if you're good with this version, please give it a ✔